### PR TITLE
Cleanup readme

### PR DESCRIPTION
**Pull request by axilleas**
:wave: I cleaned up the readme a bit. Notably:

- I removed whitespace and added newlines after headings
- I used doctoc to update the table of contents
- I got the flags in alphabetical order

Cheers!

### CODE REVIEW

**Comment by dlorenc** [comment]
Sorry for the delay, it looks like this needs a rebase :(

Ping me when you get it updated and we'll merge ASAP.

File: README.md [hunk 1]
```diff
@@ -12,36 +12,62 @@ This enables building container images in environments that can't easily or secu
 kaniko is meant to be run as an image, `gcr.io/kaniko-project/executor`.
 We do **not** recommend running the kaniko executor binary in another image, as it might not work.
 
-- [Kaniko](#kaniko)
-  - [How does kaniko work?](#how-does-kaniko-work)
-  - [Known Issues](#known-issues)
+_If you are interested in contributing to kaniko, see [DEVELOPMENT.md](DEVELOPMENT.md) and [CONTRIBUTING.md](CONTRIBUTING.md)._
```
Comment (axilleas) [inline README.md]: I moved this at the top, before the ToC to be more apparent.

File: README.md [hunk 2]
```diff
@@ -12,36 +12,62 @@ This enables building container images in environments that can't easily or secu
...
 - [Comparison with Other Tools](#comparison-with-other-tools)
 - [Community](#community)
 - [Limitations](#limitations)
+  - [mtime and snapshotting](#mtime-and-snapshotting)
 
-_If you are interested in contributing to kaniko, see [DEVELOPMENT.md](DEVELOPMENT.md) and [CONTRIBUTING.md](CONTRIBUTING.md)._
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
-### How does kaniko work?
+## How does kaniko work?
```
Comment (axilleas) [inline README.md]: Made this an H2 to follow right after the H1.

File: README.md [hunk 3]
```diff
@@ -12,36 +12,62 @@ This enables building container images in environments that can't easily or secu
...
 
 The kaniko executor image is responsible for building an image from a Dockerfile and pushing it to a registry.
 Within the executor image, we extract the filesystem of the base image (the FROM image in the Dockerfile).
 We then execute the commands in the Dockerfile, snapshotting the filesystem in userspace after each one.
 After each command, we append a layer of changed files to the base image (if there are any) and update image metadata.
 
-### Known Issues
+## Known Issues
```
Comment (axilleas) [inline README.md]: Made this an H2 to follow right after the H1.

**Comment by dlorenc** [comment]
Thanks!

**Pull request merged**